### PR TITLE
Fix failing notebook tests

### PR DESCRIPTION
**Pull request by melissachang**
Example failure:

https://scans.gradle.com/s/o5r6kcpudawae/tests/:runTestsWithTag/unit.WorkspaceOverride/notebooks()?top-execution=1

### CODE REVIEW

File: src/test/java/unit/GcpNotebookControlled.java [hunk 1]
```diff
@@ -273,6 +273,10 @@ private static void assertNotebookState(String resourceName, String notebookStat
    */
   static void assertNotebookState(String resourceName, String notebookState, UUID workspaceId)
       throws JsonProcessingException {
+    Preconditions.checkState(
+        !notebookState.equals("PROVISIONING"),
+        "Don't assert that state is PROVISIONING. This leads to flaky tests, because by the time you assert PROVISIONING, state might already be ACTIVE.");
```
Comment (jaycarlton) [inline src/test/java/unit/GcpNotebookControlled.java:278]: I'm confused. Wouldn't it also be flaky to assert that it's not provisioning anymore?

**Review by jaycarlton — COMMENTED** [review]
Thanks. I had this on my flake list.

File: src/test/java/unit/GcpNotebookControlled.java (hunk 1 above)
Comment (melissachang) [reply src/test/java/unit/GcpNotebookControlled.java]: That's a good point. I deleted this and instead made `assertNotebookState` private. So tests should only call `pollDescribeForNotebookState`, which isn't flaky.

**Review by melissachang — COMMENTED** [review]
> Thanks. I had this on my flake list.

Yeah, I saw you reran last week and it passed. For both Verily and Broad, it failed the last 3 days. I guess GCP got faster. :)

File: src/test/java/unit/GcpNotebookControlled.java [hunk 2]
```diff
@@ -284,9 +283,7 @@ static void assertNotebookState(String resourceName, String notebookState, UUID
                 "--name=" + resourceName,
                 "--workspace=" + workspaceId);
     assertEquals(notebookState, describeNotebook.state, "notebook state matches");
-    if (!notebookState.equals("PROVISIONING")) {
```
Comment (yuhuyoyo) [inline src/test/java/unit/GcpNotebookControlled.java:287]: can you keep this? since proxy url shouldn't exist for PROVISIONING state.

Comment (melissachang) [reply src/test/java/unit/GcpNotebookControlled.java]: Oops, done

**Review by yuhuyoyo — APPROVED** [review]

**Review by jaycarlton — APPROVED** [review]
Another approach would be to have the assertion take in a set of states, so you could assert that it's either PROVISIONING or ACTIVE.

**Comment by melissachang** [comment]
> Another approach would be to have the assertion take in a set of states, so you could assert that it's either PROVISIONING or ACTIVE.

Thanks. I think it's a little cleaner to say "Here's the final state I care about". It doesn't matter what the intermediate states are; someone could add a new intermediate state in the future.

**Pull request merged**